src/assignments/scrabble-scorer.rst

```diff
@@ -0,0 +1,157 @@
+Assignment #2: Scrabble Scorer
+==============================
+
+To see how far you skills have come, we are going to give you a mission to update
+our *super* important Scrabble Scorer program. What, did you think you were going
+to work on Mars rover code already?
+
+The current Scrabble Scorer program will take in a word and return the Scrabble value for
+that word. Below are the requirements for the new version.
+
+
+Requirements
+------------
+
+#. Create an ``oldScoreKey`` object that contains the previous scoring system.
+#. Create a ``transform`` function which will return a ``newScoreKey`` object.
+#. Create ``scoringAlgorithms`` array to hold three scoring objects, which include the scoring functions.
+#. Create a ``initialPrompt`` function that asks user what score algorithm to use.
+#. Create a ``runProgram`` function to contain the starting point for your running program.
+
+Example Output
+^^^^^^^^^^^^^^
+::
+
+   Welcome to the Scrabble score calculator. Use ctrl+c to quit.
+
+   Which scoring algorithm would you like to use?
+
+   0 - Scrabble: The traditional scoring algorithm.
+   1 - Simple Score: Each letter is worth 1 point.
+   2 - Bonus Vowels: Vowels are 3 pts, consonants are 1pt
+   Enter 0,1,2: 0
+   Using algorithm: Scrabble
+
+   Enter a word to be scored:  LaunchCode
+   Score for 'LaunchCode': 18
+
+   Enter a word to be scored:  Rocket
+   Score for 'Rocket': 12
+
+   Enter a word to be scored:
+
+
+Detailed Requirements
+---------------------
+
+Transform
+^^^^^^^^^
+Currently our Scrabble Grader uses this data structure when calculating scores
+using the standard Scrabble scoring algorithm.
+
+.. sourcecode:: js
+   :linenos:
+
+   const oldScoreKey = {
```

Comment:
Should we have a "space" option for the blank tiles? ``0: ' '``

Reply:
I'll add it as a bonus mission.